contrib/check-config.sh

```diff
@@ -191,9 +191,9 @@ flags=(
 	CGROUPS CGROUP_CPUACCT CGROUP_DEVICE CGROUP_FREEZER CGROUP_SCHED CPUSETS MEMCG
 	KEYS
 	VETH BRIDGE BRIDGE_NETFILTER
-	NF_NAT_IPV4 IP_NF_FILTER IP_NF_TARGET_MASQUERADE
+	"(NF_NAT_IPV4|IP_NF_NAT)" IP_NF_FILTER IP_NF_TARGET_MASQUERADE
 	NETFILTER_XT_MATCH_{ADDRTYPE,CONNTRACK,IPVS}
-	IP_NF_NAT NF_NAT NF_NAT_NEEDED
+	IP_NF_NAT NF_NAT "(NF_NAT_NEEDED|NF_NAT)"
```

Comment:
Can you elaborate on the purpose/intention of these two changes?

Reply:
I guess https://github.com/moby/moby/issues/39313#issuecomment-498497328 is what these are trying to resolve differently?

Reply:
(In a first reading, it strikes me as odd that we test for `NF_NAT` followed by another test for essentially `NF_NAT` but after reading https://github.com/moby/moby/issues/39313#issuecomment-498497328 it makes slightly more sense.)

Reply:
I'm not a fan of this because it means that if we're on a kernel which requires `NF_NAT_NEEDED`, but we've already got `NF_NAT` set, we will get a green passing result and no clues as to what we're missing.

Reply:
Yah that should have been 

IP_NF_NAT "(NF_NAT_NEEDED|NF_NAT)"

Is there really a specific kernel version where both of those are required? ( because it seems that many kernels only have one or the other) 

NF_NAT_NEEDED -> NF_NAT (>5.1)

    https://cateee.net/lkddb/web-lkddb/NF_NAT_NEEDED.html
    https://cateee.net/lkddb/web-lkddb/NF_NAT.html